spec/compiler/codegen/closure_spec.cr

```diff
@@ -681,4 +681,51 @@ describe "Code gen: closure" do
       f1.call &+ f2.call
     ))
   end
+
+  it "ensures it can raise from the closure check" do
+    expect_raises(Exception, /NoReturn/) do
```

Comment:
Can you compare with the entire expected error message here? It helps so much understanding the purpose of the spec.
Also Regex is unnecessary, String is fine.